### PR TITLE
check for duplicate candidates in block headers

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
The proofs code will be checking this too (as of https://github.com/filecoin-project/rust-fil-proofs/pull/1013) but its good to check here too.

**Comment by sternhenri** [comment]
as is mentioned in spec :D https://github.com/filecoin-project/specs/pull/783

**Comment by whyrusleeping** [comment]
wow, a readable and helpful spec PR. Thanks @sternhenri ! Youre my hero!

**Comment by whyrusleeping** [comment]
(too bad i didnt see that before I wrote this and poked @porcuquine with a number of questions)

File: chain/sync.go [hunk 1]
```diff
@@ -545,7 +545,13 @@ func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) err
 			return xerrors.Errorf("no candidates")
 		}
 
+		wins := make(map[uint64]bool)
```
Comment (Kubuxu) [inline chain/sync.go]: One small nit, is we can use `map[uint64]struct{}`.

Comment (whyrusleeping) [reply chain/sync.go]: but results in slightly cleaner code to use the bool

**Pull request merged**